x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -83,49 +77,44 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
     releasePhase = 'ga',
     getFilesClient,
   },
+  queryClient = casesQueryClient,
 }) => {
   const { appId, appTitle } = useApplication();
   const [state, dispatch] = useReducer(casesContextReducer, getInitialCasesContextState());
-  const [value, setValue] = useState<CasesContextStateValue>(() => ({
-    externalReferenceAttachmentTypeRegistry,
-    persistableStateAttachmentTypeRegistry,
-    owner,
-    permissions,
-    basePath,
-    /**
-     * The empty object at the beginning avoids the mutation
-     * of the DEFAULT_FEATURES object
-     */
-    features: merge<object, CasesFeaturesAllRequired, CasesFeatures>(
-      {},
-      DEFAULT_FEATURES,
-      features
-    ),
-    releasePhase,
-    dispatch,
-  }));
-
-  /**
-   * Only update the context if the nested permissions fields changed, this avoids a rerender when the object's reference
-   * changes.
-   */
-  useDeepCompareEffect(() => {
-    setValue((prev) => ({ ...prev, permissions }));
-  }, [permissions]);
-
-  /**
-   * `appId` and `appTitle` are dynamically retrieved from kibana context.
-   * We need to update the state if any of these values change, the rest of props are never updated.
-   */
-  useEffect(() => {
-    if (appId && appTitle) {
-      setValue((prev) => ({
-        ...prev,
-        appId,
-        appTitle,
-      }));
-    }
-  }, [appTitle, appId]);
+
+  const value = useMemo(
+    () => ({
+      appId,
+      appTitle,
+      externalReferenceAttachmentTypeRegistry,
+      persistableStateAttachmentTypeRegistry,
+      owner,
+      permissions,
+      basePath,
+      /**
+       * The empty object at the beginning avoids the mutation
+       * of the DEFAULT_FEATURES object
+       */
+      features: merge<object, CasesFeaturesAllRequired, CasesFeatures>(
+        {},
+        DEFAULT_FEATURES,
+        features
+      ),
+      releasePhase,
+      dispatch,
+    }),
+    [
+      appId,
+      appTitle,
+      basePath,
+      externalReferenceAttachmentTypeRegistry,
+      features,
+      owner,
+      permissions,
```

Comment:
Did you end up checking if this was fine for the permissions? `useMemo` does a shallow compare.

Reply:
I think is fine as it is at the moment. If we see unnecessary renders we can optimize further.

Reply:
I am ok with leaving it at is but just wanted to highlight that

> If we see unnecessary renders we can optimize further.

The issue is rather the permission changing in a way that does **not** trigger a rerender when it should be cause we only look for superficial changes to the permissions object.

Reply:
Done in [`a70ae15` (#176863)](https://github.com/elastic/kibana/pull/176863/commits/a70ae15992ba09f1412e98081db55e6b47f1c65d)